### ISSUE TITLE
Test regression: //sw/device/silicon_creator/rom/e2e:epmp_init_otp_rma_fpga_cw310_rom

### ISSUE THREAD

**Issue by dmcardle** [lowRISC/opentitan#16828]
Looks like the PROD variant of this test is timing out during bootstrapping. Some of its siblings may be failing, too, but I haven't checked yet. I bisected over cached bitstreams and got a few candidates, but I think the real culprit is f51787bb83cadc103467b6134e86504f58aa6ab1.

@tjaychen: does this seem plausible to you?

## Test output

```console
./bazelisk.sh test --test_output=streamed //sw/device/silicon_creator/rom/e2e:epmp_init_otp_rma_fpga_cw310_rom
...
[2022-12-14T18:52:32Z INFO  opentitantool::command::load_bitstream] Loading bitstream: "hw/bitstream/gcp_spliced_rom_otp_rma.bit"
[2022-12-14T18:52:32Z INFO  opentitanlib::util::usr_access] Bitstream file USR_ACCESS value: 0xfce11998
[2022-12-14T18:52:32Z INFO  opentitanlib::util::rom_detect] Current bitstream: "ROM:f32bad4b\r"
[2022-12-14T18:52:32Z INFO  opentitanlib::transport::cw310] Programming the FPGA bitstream.
[2022-12-14T18:52:46Z INFO  opentitantool] Command result: success.
[2022-12-14T18:52:46Z INFO  opentitanlib::bootstrap] Asserting bootstrap pins...
[2022-12-14T18:52:46Z INFO  opentitanlib::app] Asserting the reset signal
[2022-12-14T18:52:46Z INFO  opentitanlib::app] Clearing the UART RX buffer
[2022-12-14T18:52:46Z INFO  opentitanlib::app] Deasserting the reset signal
[2022-12-14T18:52:46Z INFO  opentitanlib::bootstrap] Performing bootstrap...
-- Test timed out at 2022-12-14 18:53:32 UTC --
```

## Bisect results

```console
% ./bazelisk.sh run //util/fpga:bitstream_bisect -- --good 6ac6849fb852d7f171f225e0350dd222cc08a12d --bad origin/master --fast-command './bazelisk.sh test --test_output=streamed //sw/device/silicon_creator/rom/e2e:epmp_init_otp_rma_fpga_cw310_rom'
Canonicalized good revision: 6ac6849fb852d7f171f225e0350dd222cc08a12d
Canonicalized bad revision:  5d721011d1fe9a49b93d52c5ce93f8e364ec566d
...
There are only 'skip'ped commits left to test.
The first bad commit could be any of:
51fb9e2f8910d44fbbbf935cfbd12c8ba0cd890b
5752f37fd48e5f2acdcf1ee4240fe0f2658be065
b4052d0e5daac2412c482dcdd24e89dec7cd6339
bd49b0fea9916776e499904519cfc16a6c084551
27610d647ebf1ae5cf89220f0363958333d1c23d
2e8f97a02e05a9de7dbb1c01fc57ead02f9b09b8
4d64e87d5d20f6165d5454ba51e395ec92412329
65966045dcca1e980ab47af9531bf0e20f7e3db1
f00c2cc169d8c87a7a4b33cd53b4901ace0eb45a
f51787bb83cadc103467b6134e86504f58aa6ab1
3d5354621e98c5824e6a72f1ace9372d6a2c4a0e
We cannot bisect more!
```

Here's the list of suspects, copied-and-pasted so GitHub will linkify them.

* 51fb9e2f8910d44fbbbf935cfbd12c8ba0cd890b
* 5752f37fd48e5f2acdcf1ee4240fe0f2658be065
* b4052d0e5daac2412c482dcdd24e89dec7cd6339
* bd49b0fea9916776e499904519cfc16a6c084551
* 27610d647ebf1ae5cf89220f0363958333d1c23d
* 2e8f97a02e05a9de7dbb1c01fc57ead02f9b09b8
* 4d64e87d5d20f6165d5454ba51e395ec92412329
* 65966045dcca1e980ab47af9531bf0e20f7e3db1
* f00c2cc169d8c87a7a4b33cd53b4901ace0eb45a
* f51787bb83cadc103467b6134e86504f58aa6ab1 <--
* 3d5354621e98c5824e6a72f1ace9372d6a2c4a0e

**Comment by tjaychen** [comment]
hmmm... that change i made should not impact the default behavior unless we actually touch those bits though.
So without that merge the PROD case works just fine?

**Comment by tjaychen** [comment]
hmm.. one thing that's suspicious about that PR is that `resval` is defined twice. Let me fix that first.

**Comment by dmcardle** [comment]
I only tested cached bitstreams, so there's a bit of ambiguity. If `bitstream_bisect` is to be believed, the test passes on dc6529c89b62ff41db5dcb4e099a5050637bbc68 and fails on 3d5354621e98c5824e6a72f1ace9372d6a2c4a0e.

```
                                              Bitstream Bisect Status: Only testing commits with cached bitstreams
┏━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━┳━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━┳━━━━━━━━┳━━━━━━━━━━┓
┃ Commit                                   ┃ Description                                                                                   ┃ Cached ┃ Judgment ┃
┡━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━╇━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━╇━━━━━━━━╇━━━━━━━━━━┩
│ 5d721011d1fe9a49b93d52c5ce93f8e364ec566d │ [bazel] Increase definition of eternal                                                        │ Cached │ BAD      │
│ 005e597980cb4b769175fda7abed036803ba6787 │ [i2c,dv] target_acq_fifo_reset                                                                │ Cached │          │
│ 5bd6c5408f815a2dca1d5652a587b930bad2d677 │ [hmac,dv] add bus integrity test                                                              │ Cached │          │
│ 3d5354621e98c5824e6a72f1ace9372d6a2c4a0e │ [ci] Change remote_timeout to 5                                                               │ Cached │          │
│ f00c2cc169d8c87a7a4b33cd53b4901ace0eb45a │ Update llvm_clang_rt_profile to llvm/llvm-project@75e33f71c2da                                │        │ SKIP     │
│ 65966045dcca1e980ab47af9531bf0e20f7e3db1 │ [sw/vendor] Update llvm_clang_rt_profile patch for llvmorg-13.0.1                             │        │ SKIP     │
│ 27610d647ebf1ae5cf89220f0363958333d1c23d │ [sw/vendor] Update llvm_clang_rt_profile.vendor.hjson                                         │        │ SKIP     │
│ bd49b0fea9916776e499904519cfc16a6c084551 │ [test] Add device_profile_data_test                                                           │        │ SKIP     │
│ 51fb9e2f8910d44fbbbf935cfbd12c8ba0cd890b │ [coverage] Improve coverage buffer printing                                                   │        │ SKIP     │
│ 5752f37fd48e5f2acdcf1ee4240fe0f2658be065 │ [coverage, ottf] Send coverage buffer regardless of the device type                           │        │ SKIP     │
│ b4052d0e5daac2412c482dcdd24e89dec7cd6339 │ [coverage] Rename test_coverage_send_buffer() to coverage_send_buffer()                       │        │ SKIP     │
│ 2e8f97a02e05a9de7dbb1c01fc57ead02f9b09b8 │ [coverage, ottf] Keep symbols in .init_array.*                                                │        │ SKIP     │
│ 4d64e87d5d20f6165d5454ba51e395ec92412329 │ [coverage, bazel] Add measure_coverage_on_target config setting                               │        │ SKIP     │
│ f51787bb83cadc103467b6134e86504f58aa6ab1 │ [flash_ctrl] Add options to change HW read behavior on info changes.                          │        │ SKIP     │
│ dc6529c89b62ff41db5dcb4e099a5050637bbc68 │ [chip testplan] Map ROM E2E debug tests                                                       │ Cached │          │
│ f3a21fc7339ae5ca9883a93392435d187811f332 │ [ci] Move SW Build and Test job to ci-public                                                  │        │ SKIP     │
│ 1c49f781560f01fa46477f089f632fd3f9b2ccdf │ [dv/scoreboard] Create function to reset the alert state                                      │        │ SKIP     │
│ af4a0d4fa326109a748be5f3ada51345ca1e2cda │ [test] Randomize the I2C target mask                                                          │ Cached │          │
│ d0b9c016fc63bbfa57bdfd1ec088b3dc475e81f7 │ [sw/silicon_creator] Add unit tests for `boot_data_read`                                      │ Cached │          │
```

**Comment by tjaychen** [comment]
i fixed the typo, but somewhat expected, it didn't actually change anything in the design
I'll run a sim and double check.

**Comment by dmcardle** [comment]
@alphan Is there any way that #16762 could be related?

**Comment by dmcardle** [comment]
Update: it's not just the PROD variant that fails, it's DEV, PROD_END, PROD, RMA, and TEST_UNLOCKED0. All of them pass at 6ac6849fb852d7f171f225e0350dd222cc08a12d and dc6529c89b62ff41db5dcb4e099a5050637bbc68.

Curiously, the RMA variant does seem to have [passed on CI](https://dev.azure.com/lowrisc/opentitan/_build/results?buildId=105453&view=logs&j=235242a5-09d5-5a5d-1aef-2327d25103b7&t=586edf67-74b1-55d6-54e3-93b6515d7956&l=17429) at 5d721011d1fe9a49b93d52c5ce93f8e364ec566d.

**Comment by dmcardle** [comment]
At @drewmacrae's suggestion, I tried `bazel clean --expunge`, and now `:epmp_init_otp_rma_fpga_cw310_rom`  passes on 5d721011d1fe9a49b93d52c5ce93f8e364ec566d. (Before expunging, I also saw that `epmp_init_otp_rma_sim_verilator` passed.)

I think this implies that I was bitten by some non-hermeticity in our build. I'm not sure how to begin debugging this. 

This implies to me there is some non-hermeticity in our build. Closing this issue, since it is vacuously solved. Thanks, @tjaychen!
